sympy/solvers/solvers.py

```diff
@@ -907,6 +907,8 @@ def solve(f, *symbols, **flags):
 
     def _sympified_list(w):
         return list(map(sympify, w if iterable(w) else [w]))
+    # denest one-element iterables. See issue #15415
+    while iterable(f) and len(f) == 1: f = next(iter(f))
```

Comment:
Could this be handled using an if condition?

This also needs test(s) to be included in `test_solvers.py`.